spec/migrations/20170928202851_migrate_middleware_server_to_wildfly_and_eap_spec.rb

```diff
@@ -0,0 +1,54 @@
+require_migration
+
+describe MigrateMiddlewareServerToWildflyAndEap do
+  let(:mw_server_stub) { migration_stub(:MiddlewareServer) }
+
+  migration_context :up do
+    it "sets new type for WildFly server" do
+      mw_server = mw_server_stub.create!(
+        :type    => "ManageIQ::Providers::Hawkular::MiddlewareManager::MiddlewareServer",
+        :product => "WildFly Full"
+      )
+      migrate
+      expect(mw_server.reload).to have_attributes(
+        :type => "ManageIQ::Providers::Hawkular::MiddlewareManager::MiddlewareServerWildfly"
+      )
+    end
+
+    it "sets new type for EAP server" do
+      mw_server = mw_server_stub.create!(
+        :type    => "ManageIQ::Providers::Hawkular::MiddlewareManager::MiddlewareServer",
+        :product => "JBoss EAP"
+      )
+      migrate
+      expect(mw_server.reload).to have_attributes(
+        :type => "ManageIQ::Providers::Hawkular::MiddlewareManager::MiddlewareServerEap"
+      )
+    end
+
+    it "leave type unchanged for other mw servers" do
+      mw_server = mw_server_stub.create!(
+        :type    => "ManageIQ::Providers::Hawkular::MiddlewareManager::MiddlewareServer",
+        :product => "Hawkular"
+      )
+      migrate
+      expect(mw_server.reload).to have_attributes(
+        :type => "ManageIQ::Providers::Hawkular::MiddlewareManager::MiddlewareServer"
+      )
+    end
+  end
+
+  migration_context :down do
+    it "reverts type for WildFly" do
+      mw_server = mw_server_stub.create!(:type => "ManageIQ::Providers::Hawkular::MiddlewareManager::MiddlewareServerWildfly")
```

Comment:
Both local assignments should be extracted to `let`, and the migration to be run in a `before` filter.

Reply:
Can you please point me to an example? I can't find any other migration which would be using it... Or maybe I just don't understand the suggestion.

Reply:
@cfcosta, does it make any difference here to move two assignments from test cases into two `let`s?

Reply:
Just checked on other migration tests, it seems like all of them are done the way you did them, so :ship: it.